openmls/src/group/mls_group/commit_builder.rs

```diff
@@ -345,6 +376,27 @@ impl<'a, G: BorrowMut<MlsGroup>> CommitBuilder<'a, Initial, G> {
 }
 
 impl<'a, G: BorrowMut<MlsGroup>> CommitBuilder<'a, LoadedPsks, G> {
+    /// Adds a PreSharedKey proposal for the provided [`PreSharedKeyId`]s to the list of proposals
+    /// to be committed.
+    ///
+    /// Note that this should not be used for sub-group branching, as those PSKs
+    /// are not allowed in regular proposals. Please use [`Self::branch()`] instead.
+    pub fn propose_psks(mut self, psk_ids: impl IntoIterator<Item = PreSharedKeyId>) -> Self {
+        self.stage.own_proposals.extend(
+            psk_ids
+                .into_iter()
+                .map(|psk_id| Proposal::psk(PreSharedKeyProposal::new(psk_id))),
+        );
+        self
+    }
+
+    /// Branches the group by adding a branch PreSharedKey proposals.
+    pub fn branch(mut self, psk_id: PreSharedKeyId) -> Self {
+        self.stage.branch = true;
+        self = self.propose_psks([psk_id]);
+        self
+    }
+
```

Comment:
I don't think these should be here. The LoadedPsk phase is too late to add more proposals, especially PSK proposals, because then these are not loaded anymore and the later computations won't have the keys (just the IDs).

Reply:
Oh yeah, this bit shouldn't be here.